src/libraries/System.Formats.Nrbf/src/System/Formats/Nrbf/BinaryArrayRecord.cs

```diff
@@ -138,7 +138,7 @@ internal static ArrayRecord Decode(BinaryReader reader, RecordMap recordMap, Pay
             lengths[i] = ArrayInfo.ParseValidArrayLength(reader);
             totalElementCount *= lengths[i];
 
-            if (totalElementCount > uint.MaxValue)
+            if (totalElementCount > ArrayInfo.MaxArrayLength)
```

Comment:
@JeremyKuhne this is a limitation I was supposed to introduce a long time ago: just don't allow for multi-dimensional arrays with the number of elements being `> Array.MaxLength`

Reply:
Should `totalElementCount` be `checked`?

Reply:
https://github.com/dotnet/runtime/blob/a3365e4c442a728bc82e59d5583cf83bc6d31d72/src/libraries/System.Formats.Nrbf/src/System/Formats/Nrbf/BinaryArrayRecord.cs#L135-L139

https://github.com/dotnet/runtime/blob/a3365e4c442a728bc82e59d5583cf83bc6d31d72/src/libraries/System.Formats.Nrbf/src/System/Formats/Nrbf/ArrayInfo.cs#L46-L53

In worst case scenario, `totalElementCount` will be `Array.MaxLength - 1` and `lengths[i]` will be `Array.MaxLength`.

It does not overflow `Int64`:

```log
> checked((long)(Array.MaxLength - 1) * (long)Array.MaxLength)
4611685771466771690
```